### ISSUE TITLE
failed to find requested hint files only on lts-12.*

### ISSUE THREAD

**Issue by leshow** [haskell/haskell-ide-engine#872]
```bash
2018-10-12 17:40:45.715944045 [ThreadId 4] - Current directory:/home/leshow/dev/haskell/haskell-programming-book/ProgrammingHaskell_Hutton
2018-10-12 17:40:45.742631268 [ThreadId 13] - Using stack GHC version
Using hie version: Version 0.2.2.0, Git revision 5d1d825880196eddbca4b1d12a60713f247c2a64 (1961 commits) x86_64 ghc-8.4.3
2018-10-12 17:40:45.882557696 [ThreadId 13] - requestDiagnostics: no diagFunc for:DiagnosticOnOpen
hie-8.4.3: user error (Failed to find requested hint files:
  /usr/bin/data/hlint.yaml
)
```

`/usr/bin/data` does not exist

This error only happens on 12.x LTS, 11.9 and others work fine.

I've installed from `haskell-ide-engine-git` in the AUR. Does anyone else have this issue? Any ideas?

**Comment by owickstrom** [comment]
Had the same issue, but where it was looking in `~/.stack/compiler-tools` for the hint file. Deleted the hie executables from the `compiler-tools` directory and reinstalled, which solved it for me.

**Comment by leshow** [comment]
@owickstrom I'll give that a shot thanks. Are you saying it was looking in `~/.stack/compiler-tools` instead of `/usr/bin/data` though ?

**Comment by owickstrom** [comment]
@leshow Yeah, exactly.

**Comment by leshow** [comment]
I don't have a `~/.stack/compiler-tools`, `locate` can't find any folder by that name either.

**Comment by leshow** [comment]
Ah okay, it appears (after removing hie) that there were a bunch of binaries still in `~/.local/bin` I've just removed these, and am re-installing hie. I'll close the thread if that fixes things

**Comment by leshow** [comment]
I re-installed and have now broken all of hie. I get the same error on 12.x LTS and on 11.x or lower I get:

```
hie-8.2.2: Could not find $libexecdir/cabal-helper-wrapper

If you are a cabal-helper developer you can set the environment variable
`cabal_helper_libexecdir' to override $libexecdir[1]. The following will
work in the cabal-helper source tree:

    $ export cabal_helper_libexecdir=$PWD/dist/build/cabal-helper-wrapper

[1]: /var/cache/pacman/pkg/haskell-ide-engine-git19859/haskell-ide-engine-git/src/haskell-ide-engine/.stack-work/install/x86_64-linux-tinfo6/lts-11.18/8.2.2/libexec/x86_64-linux-ghc-8.2.2/cabal-helper-0.8.1.0

If you don't know what I'm talking about something went wrong with your
installation. Please report this problem here:

    https://github.com/DanielG/cabal-helper/issues
```

**Comment by bubba** [comment]
A quick workaround:

```bash
mkdir ~/.local/bin/data
touch ~/.local/bin/data/hlint.yaml
```

I think it's weird that hlint throws an exception when there's no global file. It could be hie messing up when its invoking it

**Comment by alanz** [comment]
TODO: update README

**Comment by jarcane** [comment]
I am having this same problem. It complains about the missing hlint file, and if I do the workaround @bubba suggests I get a similar error to @leshow :

```
hie-8.6.3: Could not find $libexecdir/cabal-helper-wrapper

If you are a cabal-helper developer you can set the environment variable
`cabal_helper_libexecdir' to override $libexecdir[1]. The following will
work in the cabal-helper source tree:

    $ export cabal_helper_libexecdir=$PWD/dist/build/cabal-helper-wrapper

[1]: /home/annaia/projects/haskell-ide-engine/.stack-work/install/x86_64-linux-tinfo6/lts-13.0/8.6.3/libexec/x86_64-linux-ghc-8.6.3/cabal-helper-0.9.0.0
/home/annaia/projects/haskell-ide-engine/.stack-work/install/x86_64-linux-tinfo6/lts-13.0/8.6.3/libexec/x86_64-linux-ghc-8.6.3/cabal-helper-0.9.0.0/cabal-helper-wrapper
junk/build/cabal-helper-wrapper/cabal-helper-wrapper
/home/annaia/projects/haskell-ide-engine/.stack-work/install/x86_64-linux-tinfo6/lts-13.0/8.6.3/bin/cabal-helper-wrapper

If you don't know what I'm talking about something went wrong with your
installation. Please report this problem here:

    https://github.com/DanielG/cabal-helper/issues
```

hie --version: Version 0.5.0.0, Git revision c4a9e7cfce86194493834839271c2aa390483048 (2382 commits) x86_64 ghc-8.6.3

**Comment by alanz** [comment]
@jarcane the current version of hie is 0.6.0.0, and it did update hlint.

Please re-install and try again.

**Comment by LinArcX** [comment]
Still same issue with arch linux and install bunch of packages like __hoogle__, __hlint__. but still complain about hlint.yaml. I follow @owickstrom and @bubba . but nothing changed.

**Comment by leshow** [comment]
@LinArcX don't install any other packages from AUR, afaik hie provides its own versions of the required packages

**Comment by LinArcX** [comment]
@leshow:
I installed hie binary version from archlinuxcn. 
Is it OK?

**Comment by bubba** [comment]
> ```
> hie-8.6.3: user error (Failed to find requested hint files:
>   /usr/bin/data/hlint.yaml
> )
> ```

This is one of the issues we're having with generating our own static binaries: [hlint relies on data-files which currently there's not a great way of bundling with a binary](https://github.com/haskell/haskell-ide-engine/issues/1052#issuecomment-454598200), and similarly for cabal-helper
For now though `touch /usr/bin/data/hlint.yaml` should fix it 😬

**Comment by LinArcX** [comment]
@bubba nothing changed.

**Comment by fendor** [comment]
This should be different now, hie does no longer choke on that, however, hlint does not work if there is no hlint.yaml, or in general, when its data-files are missing. The issue #1143 aims to solve this problem as a whole, once we have decision for it.

**Comment by jneira** [comment]
Closed by #1143: hlint data files could be placed in a subfolder `data` in the same directory where the `hie` executable is. We'll enforce that in the release artifacts.

Only left update docs.
